diff -ur php52-5.2.14/configure php52-5.2.14-recode/configure
--- php52-5.2.14/configure	2010-07-21 16:31:30.000000000 +0200
+++ php52-5.2.14-recode/configure	2010-10-28 13:59:48.186113053 +0200
@@ -107200,7 +107200,7 @@
   test "$PHP_MYSQL" != "no" && recode_conflict="$recode_conflict mysql"
 
   if test -n "$recode_conflict"; then
-    { echo "configure: error: recode extension can not be configured together with:$recode_conflict" 1>&2; exit 1; }
+    echo "configure: warning: recode extension can not be configured together with:$recode_conflict" 1>&2
   fi
 fi
 
diff -ur php52-5.2.14/ext/recode/config9.m4 php52-5.2.14-recode/ext/recode/config9.m4
--- php52-5.2.14/ext/recode/config9.m4	2004-11-09 17:11:13.000000000 +0100
+++ php52-5.2.14-recode/ext/recode/config9.m4	2010-10-28 13:57:37.776129240 +0200
@@ -8,6 +8,6 @@
   test "$PHP_MYSQL" != "no" && recode_conflict="$recode_conflict mysql"
 
   if test -n "$recode_conflict"; then
-    AC_MSG_ERROR([recode extension can not be configured together with:$recode_conflict])
+    AC_MSG_WARN([recode extension can not be configured together with:$recode_conflict])
   fi
 fi
