diff -urN tartarus-0.9.4/bin/charon.ftp tartarus-0.9.4-ftptruncate/bin/charon.ftp
--- tartarus-0.9.4/bin/charon.ftp	2010-09-11 01:27:05.000000000 +0200
+++ tartarus-0.9.4-ftptruncate/bin/charon.ftp	2011-01-05 16:37:43.147952568 +0100
@@ -27,6 +27,7 @@
 my $all = 0;
 
 my $dry_run = 0;
+my $truncate = 0;
 my $verbose = 0;
 my $help = 0;
 
@@ -38,7 +39,7 @@
 
 =head1 SYNOPSIS
 
-B<charon.ftp> --host I<host> [--ssl] --user I<name> (--password I<password> | --readpassword) [--dir F<directory>] --maxage I<days> (--profile I<identifier> | --all) [--dry-run] [--verbose] [--help]
+B<charon.ftp> --host I<host> [--ssl] --user I<name> (--password I<password> | --readpassword) [--dir F<directory>] --maxage I<days> (--profile I<identifier> | --all) [--truncate] [--dry-run] [--verbose] [--help]
 
 =head1
 
@@ -90,6 +91,12 @@
 When this is set, the expiration date is applied to alle files found on the server
 and identified as tartarus backups.
 
+=item B<--truncate>
+
+This option enables replacing files with /dev/null on FTP server before deletion.
+Some FTP servers won't allow deleting files when the quota is reached, this flag
+enables a workaround.
+
 =item B<--dry-run>
 
 This option prevents charon from actually deleting files.
@@ -121,6 +128,7 @@
     --profile       backup profile name to process
     --all           process all files found in the directory
     --dry-run       do a test run, don't actually delete files
+    --truncate      truncate files on server before deleting
     --verbose       be more verbose about the actions made
     --help          show this help text
 EOF
@@ -141,6 +149,7 @@
     "all|a"             => \$all,
     "directory|dir|d=s" => \$dir,
     "test|dry-run|n"    => \$dry_run,
+    "truncate|t"        => \$truncate,
     "verbose|v"         => \$verbose,
     "help|h"            => \$help
 ) || usage();
@@ -190,6 +199,9 @@
 for my $file ($c->expire($days_to_expire, $uprofile)) {
     print STDERR "Removing file $file...\n";
     unless ($dry_run) {
+        if ($truncate) {
+            $ftp->put("/dev/null", "$file");
+        }
         $ftp->delete("$file") || print STDERR "Error removing $file!\n";
     }
 }
