diff -ur php52-apc-3.1.5/php_apc.h php52-apc-3.1.5-version/php_apc.h
--- php52-apc-3.1.5/php_apc.h	2010-11-02 20:17:30.000000000 +0100
+++ php52-apc-3.1.5-version/php_apc.h	2010-11-17 16:44:59.968071905 +0100
@@ -35,7 +35,7 @@
 #include "apc_php.h"
 #include "apc_globals.h"
 
-#define PHP_APC_VERSION "3.1.5-dev"
+#define PHP_APC_VERSION "3.1.5"
 
 extern zend_module_entry apc_module_entry;
 #define apc_module_ptr &apc_module_entry
