diff -ur php53-5.3.3/ext/session/session.c php53-5.3.3-defaults/ext/session/session.c
--- php53-5.3.3/ext/session/session.c	2010-06-09 00:30:16.000000000 +0200
+++ php53-5.3.3-defaults/ext/session/session.c	2010-10-29 15:15:05.596122060 +0200
@@ -788,7 +788,7 @@
 PHP_INI_BEGIN()
 	STD_PHP_INI_BOOLEAN("session.bug_compat_42",    "1",         PHP_INI_ALL, OnUpdateBool,   bug_compat,         php_ps_globals,    ps_globals)
 	STD_PHP_INI_BOOLEAN("session.bug_compat_warn",  "1",         PHP_INI_ALL, OnUpdateBool,   bug_compat_warn,    php_ps_globals,    ps_globals)
-	STD_PHP_INI_ENTRY("session.save_path",          "",          PHP_INI_ALL, OnUpdateSaveDir,save_path,          php_ps_globals,    ps_globals)
+	STD_PHP_INI_ENTRY("session.save_path",          "/var/lib/php53", PHP_INI_ALL, OnUpdateSaveDir,save_path,          php_ps_globals,    ps_globals)
 	STD_PHP_INI_ENTRY("session.name",               "PHPSESSID", PHP_INI_ALL, OnUpdateString, session_name,       php_ps_globals,    ps_globals)
 	PHP_INI_ENTRY("session.save_handler",           "files",     PHP_INI_ALL, OnUpdateSaveHandler)
 	STD_PHP_INI_BOOLEAN("session.auto_start",       "0",         PHP_INI_ALL, OnUpdateBool,   auto_start,         php_ps_globals,    ps_globals)
diff -ur php53-5.3.3/main/main.c php53-5.3.3-defaults/main/main.c
--- php53-5.3.3/main/main.c	2010-03-12 11:28:59.000000000 +0100
+++ php53-5.3.3-defaults/main/main.c	2010-10-29 15:15:35.666122485 +0200
@@ -521,7 +521,7 @@
 	PHP_INI_ENTRY("mail.force_extra_parameters",NULL,		PHP_INI_SYSTEM|PHP_INI_PERDIR,		OnChangeMailForceExtra)
 	PHP_INI_ENTRY("disable_functions",			"",			PHP_INI_SYSTEM,		NULL)
 	PHP_INI_ENTRY("disable_classes",			"",			PHP_INI_SYSTEM,		NULL)
-	PHP_INI_ENTRY("max_file_uploads",			"20",			PHP_INI_SYSTEM,		NULL)
+	PHP_INI_ENTRY("max_file_uploads",			"50",			PHP_INI_SYSTEM,		NULL)
 
 	STD_PHP_INI_BOOLEAN("allow_url_fopen",		"1",		PHP_INI_SYSTEM,		OnUpdateBool,		allow_url_fopen,		php_core_globals,		core_globals)
 	STD_PHP_INI_BOOLEAN("allow_url_include",	"0",		PHP_INI_SYSTEM,		OnUpdateBool,		allow_url_include,		php_core_globals,		core_globals)
diff -ur php53-5.3.3/php.ini-development php53-5.3.3-defaults/php.ini-development
--- php53-5.3.3/php.ini-development	2010-06-24 02:15:12.000000000 +0200
+++ php53-5.3.3-defaults/php.ini-development	2010-10-29 15:15:43.746122268 +0200
@@ -879,7 +879,7 @@
 upload_max_filesize = 2M
 
 ; Maximum number of files that can be uploaded via a single request
-max_file_uploads = 20
+max_file_uploads = 50
 
 ;;;;;;;;;;;;;;;;;;
 ; Fopen wrappers ;
diff -ur php53-5.3.3/php.ini-production php53-5.3.3-defaults/php.ini-production
--- php53-5.3.3/php.ini-production	2010-06-24 02:15:12.000000000 +0200
+++ php53-5.3.3-defaults/php.ini-production	2010-10-29 15:15:53.696122919 +0200
@@ -879,7 +879,7 @@
 upload_max_filesize = 2M
 
 ; Maximum number of files that can be uploaded via a single request
-max_file_uploads = 20
+max_file_uploads = 50
 
 ;;;;;;;;;;;;;;;;;;
 ; Fopen wrappers ;
