diff -ur php53-5.3.6/ext/pdo_mysql/mysql_driver.c php53-5.3.6-mysqlnd-compression/ext/pdo_mysql/mysql_driver.c
--- php53-5.3.6/ext/pdo_mysql/mysql_driver.c	2011-01-17 10:54:22.000000000 +0100
+++ php53-5.3.6-mysqlnd-compression/ext/pdo_mysql/mysql_driver.c	2011-07-01 12:30:57.822439123 +0200
@@ -700,7 +700,7 @@
 			}
 			efree(default_group);
 		}
-
+#endif
 		compress = pdo_attr_lval(driver_options, PDO_MYSQL_ATTR_COMPRESS, 0 TSRMLS_CC);
 		if (compress) {
 			if (mysql_options(H->server, MYSQL_OPT_COMPRESS, 0)) {
@@ -708,7 +708,6 @@
 				goto cleanup;
 			}
 		}
-#endif
 	}
 
 #ifdef PDO_MYSQL_HAS_CHARSET
diff -ur php53-5.3.6/ext/pdo_mysql/pdo_mysql.c php53-5.3.6-mysqlnd-compression/ext/pdo_mysql/pdo_mysql.c
--- php53-5.3.6/ext/pdo_mysql/pdo_mysql.c	2011-01-01 03:19:59.000000000 +0100
+++ php53-5.3.6-mysqlnd-compression/ext/pdo_mysql/pdo_mysql.c	2011-07-01 12:31:11.712466440 +0200
@@ -74,8 +74,8 @@
 	REGISTER_PDO_CLASS_CONST_LONG("MYSQL_ATTR_MAX_BUFFER_SIZE", (long)PDO_MYSQL_ATTR_MAX_BUFFER_SIZE);
 	REGISTER_PDO_CLASS_CONST_LONG("MYSQL_ATTR_READ_DEFAULT_FILE", (long)PDO_MYSQL_ATTR_READ_DEFAULT_FILE);
 	REGISTER_PDO_CLASS_CONST_LONG("MYSQL_ATTR_READ_DEFAULT_GROUP", (long)PDO_MYSQL_ATTR_READ_DEFAULT_GROUP);
-	REGISTER_PDO_CLASS_CONST_LONG("MYSQL_ATTR_COMPRESS", (long)PDO_MYSQL_ATTR_COMPRESS);
 #endif
+	REGISTER_PDO_CLASS_CONST_LONG("MYSQL_ATTR_COMPRESS", (long)PDO_MYSQL_ATTR_COMPRESS);
 	REGISTER_PDO_CLASS_CONST_LONG("MYSQL_ATTR_DIRECT_QUERY", (long)PDO_MYSQL_ATTR_DIRECT_QUERY);
 	REGISTER_PDO_CLASS_CONST_LONG("MYSQL_ATTR_FOUND_ROWS", (long)PDO_MYSQL_ATTR_FOUND_ROWS);
 	REGISTER_PDO_CLASS_CONST_LONG("MYSQL_ATTR_IGNORE_SPACE", (long)PDO_MYSQL_ATTR_IGNORE_SPACE);
diff -ur php53-5.3.6/ext/pdo_mysql/php_pdo_mysql_int.h php53-5.3.6-mysqlnd-compression/ext/pdo_mysql/php_pdo_mysql_int.h
--- php53-5.3.6/ext/pdo_mysql/php_pdo_mysql_int.h	2011-01-17 10:54:22.000000000 +0100
+++ php53-5.3.6-mysqlnd-compression/ext/pdo_mysql/php_pdo_mysql_int.h	2011-07-01 12:31:30.532437296 +0200
@@ -164,8 +164,8 @@
 	PDO_MYSQL_ATTR_READ_DEFAULT_FILE,
 	PDO_MYSQL_ATTR_READ_DEFAULT_GROUP,
 	PDO_MYSQL_ATTR_MAX_BUFFER_SIZE,
-	PDO_MYSQL_ATTR_COMPRESS,
 #endif
+	PDO_MYSQL_ATTR_COMPRESS,
 	PDO_MYSQL_ATTR_DIRECT_QUERY,
 	PDO_MYSQL_ATTR_FOUND_ROWS,
 	PDO_MYSQL_ATTR_IGNORE_SPACE
diff -ur php53-5.3.6/ext/pdo_mysql/tests/pdo_mysql_class_constants.phpt php53-5.3.6-mysqlnd-compression/ext/pdo_mysql/tests/pdo_mysql_class_constants.phpt
--- php53-5.3.6/ext/pdo_mysql/tests/pdo_mysql_class_constants.phpt	2009-10-26 12:41:30.000000000 +0100
+++ php53-5.3.6-mysqlnd-compression/ext/pdo_mysql/tests/pdo_mysql_class_constants.phpt	2011-07-01 12:32:15.212436896 +0200
@@ -15,13 +15,13 @@
 		'MYSQL_ATTR_FOUND_ROWS'						=> true,
 		'MYSQL_ATTR_IGNORE_SPACE'					=> true,
 		'MYSQL_ATTR_INIT_COMMAND'					=> true,
+		'MYSQL_ATTR_COMPRESS'						=> true;
 	);
 
 	if (!MySQLPDOTest::isPDOMySQLnd()) {
 		$expected['MYSQL_ATTR_MAX_BUFFER_SIZE'] 		= true;
 		$expected['MYSQL_ATTR_READ_DEFAULT_FILE'] 	= true;
 		$expected['MYSQL_ATTR_READ_DEFAULT_GROUP'] 	= true;
-		$expected['MYSQL_ATTR_COMPRESS']						= true;
 	}
 
 	/*
